lib/inc/internal/facts/linux/processor_resolver.hpp

```diff
@@ -14,12 +14,40 @@ namespace facter { namespace facts { namespace linux {
     struct processor_resolver : posix::processor_resolver
     {
      protected:
+        enum ArchitectureType {POWER, X86};
+
+	/**
+	 * The check consists of the following.
+	 *   (1) Check the previously computed isa fact. If it starts with
+	 *   ppc64, then we have a power machine.
+	 *
+	 *   (2) If (1) is empty (possible because exec might have failed to obtain
+	 *   the isa fact), then we use /proc/cpuinfo by checking whether that file
+	 *   contains the "cpu", "clock", and "revision" keys -- these keys are only
+	 *   found in Power machines.
+         *
+         * @param data The currently collected data
+         * @param root Path to the root directory of the system
+	 */ 
+        ArchitectureType architecture_type(data const& data, std::string const& root);
+
+       /**
+        * Adds the cpu-specific data to the currently collected data.
+        * @param data The currently collected data
+        * @param root Path to the root directory of the system
+        */
+        void add_cpu_data(data& data, std::string const& root = "");
+
         /**
          * Collects the resolver data.
          * @param facts The fact collection that is resolving facts.
          * @return Returns the resolver data.
          */
         virtual data collect_data(collection& facts) override;
+
+     private:
+        void add_generic_cpu_data(data& data, bool have_counts, std::string const& root = "");
```

Comment:
Since you renamed the enum member from `GENERIC` to `X86`, I'd rename this function too, myself